api/rest/client.go

```diff
@@ -233,7 +234,7 @@ func (c *Client) sendWithRetries(ctx context.Context, req *http.Request, retryCo
 		c.rateLimiter.Update(ctx, response.StatusCode, response.Header)
 	}
 
-	if c.retryOptions != nil && retryCount < c.retryOptions.MaxRetries {
+	if ShouldRetry(response.StatusCode) && c.retryOptions != nil && retryCount < c.retryOptions.MaxRetries {
```

Comment:
Nit: To improve readability, I personally would move this c.retryOptions != nil && retryCount < c.retryOptions.MaxRetrie into the `ShouldRetry` method.

Reply:
My plan was to make it re-usable also for Monaco